### PR TITLE
*6625* refactor 'exclude and clear for single objects and exclude all issue object...'

### PR DESCRIPTION
**Pull request by jnugent**
This is a rebase of the original pull request for bug 6625, rebased against ojs-dev-2_4. 
Original pull request:  https://github.com/pkp/ojs/pull/217

### CODE REVIEW

File: classes/issue/IssueDAO.inc.php [hunk 1]
```diff
@@ -818,6 +818,25 @@ function deleteAllPubIds($journalId, $pubIdType) {
 	}
 
 	/**
+	 * Delete the public ID of an issues.
```
Comment (asmecher) [inline classes/issue/IssueDAO.inc.php]: an issue

File: classes/article/ArticleGalleyDAO.inc.php [hunk 2]
```diff
@@ -632,6 +632,25 @@ function deleteAllPubIds($journalId, $pubIdType) {
...
+	function deletePubId($galleyId, $pubIdType) {
+		$settingName = 'pub-id::'.$pubIdType;
+		$this->update(
+			'DELETE FROM article_galley_settings WHERE setting_name = ? AND galley_id = ?',
+			array(
+				$settingName,
+				(int)$galleyId
+			)
+		);
+		$this->flushCache();
```
Comment (asmecher) [inline classes/article/ArticleGalleyDAO.inc.php]: I don't think flushCache does what you expect in ArticleGalleyDAO (see also another use just above this one).

File: classes/article/SuppFileDAO.inc.php [hunk 3]
```diff
@@ -478,6 +478,25 @@ function deleteAllPubIds($journalId, $pubIdType) {
...
+	function deletePubId($suppFileId, $pubIdType) {
+		$settingName = 'pub-id::'.$pubIdType;
+		$this->update(
+			'DELETE FROM article_supp_file_settings WHERE setting_name = ? AND supp_id = ?',
+			array(
+				$settingName,
+				(int)$suppFileId
+			)
+		);
+		$this->flushCache();
```
Comment (asmecher) [inline classes/article/SuppFileDAO.inc.php]: As per https://github.com/pkp/ojs/pull/425/files#r24856795

**Comment by asmecher** [comment]
This looks fine to me, with a couple of small comments, but I haven't tested it. @bozana, would you be able to run a test of this in the next little while? We're already into translations for OJS 2.4.6, so it's somewhat time-sensitive. Thanks!

File: classes/article/ArticleGalleyDAO.inc.php (hunk 2 above)
Comment (asmecher) [inline classes/article/ArticleGalleyDAO.inc.php]: Did you address this, Jason?

Comment (jnugent) [inline classes/article/ArticleGalleyDAO.inc.php]: If you were referring to the fact that ArticleGalleyDAO had no specific implementation of flushCache and relied on what was in DAO, I've added a method override in ArticleGalleyDAO that specifically fetches the ArticleGalley cache and calls $cache->flush() now.  It's part of the new commit from yesterday.

Comment (asmecher) [inline classes/article/ArticleGalleyDAO.inc.php]: Thanks, but how about the same for SuppFileDAO?

Comment (jnugent) [inline classes/article/ArticleGalleyDAO.inc.php]: whoops, done!

File: classes/plugins/PubIdPlugin.inc.php [hunk 4]
```diff
@@ -339,6 +371,78 @@ function getAdditionalFieldNames($hookName, $params) {
...
+	function editIssue($hookName, $params) {
+		$issue =& $params[0];
+		$issueId = $issue->getId();
+
+		$pubIdPlugins =& PluginRegistry::loadCategory('pubIds', true);
+		if (is_array($pubIdPlugins)) {
+			foreach ($pubIdPlugins as $pubIdPlugin) {
+				$excludeSubmittName = 'excludeIssueObjects_' . $pubIdPlugin->getPubIdType();
+				$clearSubmittName = 'clearIssueObjects_' . $pubIdPlugin->getPubIdType();
+				$exlude = $clear = false;
```
Comment (asmecher) [inline classes/plugins/PubIdPlugin.inc.php]: *exclude

**Comment by asmecher** [comment]
Thanks, Jason, looks good. Just the one typo correction. I'd like to get this into the translation install ASAP, but would still like to go over it with UI/UX, so go ahead and merge, then we'll get it into the translation install, then review it with UI/UX. Kind of backwards, but I think it'll be fine.

**Pull request merged**